src/common/CMakeLists.txt

```diff
@@ -67,9 +68,15 @@ else()
     set( LIBRARY_TYPE SHARED )
 endif()
 
+if( HAVE_CLOUDSC_GPU_SCC_FIELD )
+    list(APPEND CLOUDSC_COMMON_SOURCES
+        module/cloudsc_field_state_mod.F90
+    )
+endif()
+
 ecbuild_add_library( TARGET cloudsc-common-lib
```

Comment:
Add here the link options for CUDA to the public build interface:
```cmake
if ( HAVE_FIELD_API AND HAVE_CUDA )
  target_link_options( cloudsc-common-lib PUBLIC "-cuda" )
endif()
```
https://cmake.org/cmake/help/latest/command/target_link_options.html